lib/calabash/http/retriable_client.rb

```diff
@@ -121,6 +128,13 @@ def request(request, request_method, options={})
                                request.params, header)
           rescue *RETRY_ON => e
             @logger.log "Http error: #{e}"
+
+            if first_try
+              if @on_error[e.class]
+                @on_error[e.class].call(@server)
+              end
+            end
+
             last_error = e
```

Comment:
@TobiasRoikjer Won't this trigger on iOS as well?

Shouldn't there be a branch on whether or not there is an @on_error defined?